VERSIONING.md

```diff
@@ -0,0 +1,32 @@
+# Splunk Distribution of OpenTelemetry Java Instrumentation Versioning
+
+## Version numbers
+
+The project strictly follows [Semantic Versioning 2.0.0](https://semver.org/). This means
+that all artifacts have a version of the format `MAJOR.MINOR.PATCH`.
+No backwards-incompatible changes will be made unless incrementing the `MAJOR` version number. 
+Most releases will be made by incrementing the `MINOR` version. 
+Patch releases will be made by incrementing the `PATCH` version.
+
+## Compatibility  
+
+Backwards-incompatible changes will be avoided - if they are necessary, the `MAJOR` version of the artifact will be
+incremented.
+
+This can occur in following situations:
+- OpenTelemetry Java Instrumentation breaking change - please see [upstream repository](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/VERSIONING.md) for their versioning policy
```

Comment:
It's also possible that this happens because of an upstream-upstream breaking change in OpenTelemetry-Java itself, yes? Like, if something with the autoconfiguration artifact changes in a breaking way? (especially since that's not marked stable at this point)

Reply:
Yes, but I believe that from our and customer point of view the major release will be triggered by major release of the upstream (ie if otel-java inroduces breaking change, that will be followed by otel-java agent release and then - splunk release). So I'd leave this wording to avoid confusion.

Reply:
OTel Java Instrumentation VERSIONING file itself relies on the OTel Java versioning policy.
Still, it wouldn't hurt to be a bit more explicit.